### PR TITLE
Add ReCoRD evaluation methods

### PR DESCRIPTION
**Pull request by jeswan**
This PR adds evaluation methods for ReCoRD (and closes https://github.com/jiant-dev/jiant/issues/51). These changes are required to run all SuperGLUE tasks E2E using the SuperGLUE submission formatter in https://github.com/jiant-dev/jiant/pull/142. To evaluate with ReCoRD, it is necessary to have the set of gold labels to generate the metrics. To accomplish this, the set of gold labels was added to each `DataRow`. Other solutions such as adding all examples to the`batch_metadata` were considered but added unnecessary bloat to the data stored.

**How were the changes validated**
The evaluation methods were manually inspected (they are different than what was used in jiant). The validation and test F1 scores were similar for few-shot training (val: 0.10, test: 0.14). @zphang Should I run fine-tuning on this task to further validate the changes? Also, are there any additional changes to close https://github.com/jiant-dev/jiant/issues/51?

### CODE REVIEW

**Comment by zphang** [comment]
Can we do a BERT-large-cased run? We want to get a number that's hopefully not too far off from: https://arxiv.org/pdf/1905.00537.pdf

File: jiant/tasks/evaluate/core.py [hunk 1]
```diff
@@ -406,50 +424,70 @@ def get_labels_from_examples(cls, examples):
...
-        cls, logits, examples: List[RecordLabelData]
-    ):
-        psg_qns_idx_dict = {}
-        for i, example in examples:
-            psq_qns_idx = example.passage_idx, example.question_idx
-            if psq_qns_idx not in psg_qns_idx_dict:
-                psg_qns_idx_dict[psq_qns_idx] = []
-            psg_qns_idx_dict[psq_qns_idx].append(i)
-
+    def compute_preds_and_metrics_from_logits_and_record_labels(cls, task, accumulator):
```
Comment (zphang) [inline jiant/tasks/evaluate/core.py:460]: Rename function, or make a separate function, since it's not operating off logits and labels.

**Comment by jeswan** [comment]
@zphang bert-large-cased val set results (em/f1): 64.1/64.9 compared to test set results 71.3/72.0 from https://arxiv.org/pdf/1905.00537.pdf. What are your thoughts? Should we take a look at the arguments and run again?

**Comment by zphang** [comment]
See:  #186

**Review by zphang — APPROVED** [review]

File: jiant/tasks/core.py [hunk 2]
```diff
@@ -224,3 +224,9 @@ def get_glue_preds(cls, pred_dict: Dict):
             indexes.append(int(guid.split("-")[1]))
             predictions.append(str(cls.LABELS[pred]).lower())
         return (indexes, predictions)
+
+
+class SuperGlueMixin:
+    @classmethod
+    def super_glue_format_preds(cls, pred_dict):
+        raise NotImplementedError()
```
Comment (jeswan) [inline jiant/tasks/core.py:232]:
```suggestion
```
